ctl/multi-reader.go

```diff
@@ -0,0 +1,55 @@
+package ctl
+
+// Reader wraps an io.Reader, allowing "clones" to be created via the View method.
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+type MultiReader struct {
```

Comment:
I think we should pick a different name for this, because `io.MultiReader` already exists, for a reader which combines several other readers. Maybe `ReReader`?